applications/conversations/controllers/api/ConversationsApiController.php

```diff
@@ -0,0 +1,452 @@
+
+        $conversations = $this->conversationModel->get2($userID, $offset, $limit)->resultArray();
+
+        if (!empty($query['expand'])) {
+            $this->userModel->expandUsers($conversations, ['InsertUserID']);
+        }
+
+        return $out->validate($conversations, true);
+    }
+
+    /**
+     * Get an ID-only conversation record schema.
+     *
+     * @return Schema Returns a schema object.
+     */
+    private function idParamSchema() {
+        return $this->schema(['id:i' => 'The conversation ID.'], 'in');
+    }
+
+    /**
+     * Add a conversation.
+     *
+     * @param array $body The request body.
+     * @throws ServerException If the conversation could not be created.
+     * @return Data
+     */
+    public function post(array $body) {
+        $this->permission('Conversations.Conversations.Add');
+
+        $in = $this->postSchema('in')->setDescription('Add a conversation.');
+        $out = $this->schema($this->fullSchema(), 'out');
+
+        $body = $in->validate($body);
+        $conversationData = $this->normalizeInput($body);
+        $conversationData = $this->caseScheme->convertArrayKeys($conversationData);
+
+        $conversationID = $this->conversationModel->save($conversationData, ['ConversationOnly' => true]);
+        $this->validateModel($this->conversationModel, true);
+        if (!$conversationID) {
+            throw new ServerException('Unable to insert conversation.', 500);
+        }
+
+        $conversation = $this->conversationByID($conversationID, $this->getSession()->UserID);
+        return new Data(
+            $out->validate($conversation),
+            201
+        );
+    }
+
+    /**
+     * Add participants to a conversation.
+     *
+     * @param int $id The ID of the conversation.
+     * @param array $body The request body.
+     * @throws NotFoundException if the conversation could not be found.
+     * @throws ServerException If the participants could not be added.
+     * @return Data
+     */
+    public function post_participants($id, array $body) {
+        $this->permission('Conversations.Conversations.Add');
+
+        $this->idParamSchema();
+
+        $in = $this->postSchema('in')->setDescription('Add participants to a conversation.');
+        $this->schema([], 'out');
+
+        $this->conversationByID($id);
+
+        $body = $in->validate($body);
+
+        $success = $this->conversationModel->addUserToConversation($id, $body['participantIDs']);
+        if (!$success) {
+            throw new ServerException('Unable to add participants.', 500);
+        }
+
+        return new Data([], 201);
+    }
+
+    /**
+     * Get a post schema with minimal add/edit fields.
+     *
+     * @param string $type The type of schema.
+     * @return Schema Returns a schema object.
+     */
+    public function postSchema($type) {
+        static $postSchema;
+
+        if ($postSchema === null) {
+            $inSchema = [
+                'participantIDs:a' => [
```

Comment:
We will have a problem with this if we use UserIDs everywhere. 
What will happen if there are 2 sets or UserIDs.

This might be a problem in groups. Admin vs Moderators vs Members.
We can't use UserIDs everywhere it will become more confusing than anything.
There's documentation on the endpoint with the description. That should be good enough.

Reply:
We will discuss that in our next R&D meeting.